### PR TITLE
feat(test): Install jest and add a few example test files [TINY-904]

### PR DESCRIPTION
**Pull request by jamcry**
### Description
- Install jest and type definitions
- Add two test files for starters

### TODO : Things to discuss/think about
- How to test functions that use algod client
  > I think using the real algod client for test wouldn't be efficient, maybe we can implement a mock function
- Test file/folder structure
  > I like the current structure: `a.ts, a.test.ts`

### CODE REVIEW

File: src/validator.test.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+describe("`getValidatorAppID`", () => {
+  test("throws an error when one of the args is invalid", () => {
+    expect(() =>
+      getValidatorAppID("unknown" as any, CONTRACT_VERSION.V1_1)
+    ).toThrowError();
+    expect(() => getValidatorAppID("mainnet", "unknown" as any)).toThrowError();
+  });
+
+  test("returns the correct validator app id", () => {
```
Comment (jamcry) [inline src/validator.test.ts:15]: I'm not very confident that this is a useful test 😄 Open for comments

File: package.json [hunk 2]
```diff
@@ -42,6 +43,7 @@
     "@rollup/plugin-commonjs": "^22.0.2",
     "@rollup/plugin-json": "^4.1.0",
     "@rollup/plugin-node-resolve": "^14.0.0",
+    "@types/jest": "^29.4.0",
```
Comment (jamcry) [inline package.json:46]: there were two options here `@types/jest` and `@jest/globals`. But auto import for test functions didn't work with `@jest/globals` so I used `@types`

https://jestjs.io/docs/getting-started#type-definitions

File: src/util/asset/assetUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import {describe, expect, test} from "@jest/globals";
+
+import {ALGO_ASSET_ID} from "./assetConstants";
+import {isAlgo, prepareAssetPairData, sortAssetIds} from "./assetUtils";
+
+test("`sortAssetIds` returns an array with bigger id first", () => {
```
Comment (Anlerkan) [inline src/util/asset/assetUtils.test.ts:6]: Maybe we can add additional checks to the sortAssetIds function and their tests, what do you think? Here are the checks I suggest:
```tsx
  if (asset1ID === asset2ID) {
    throw new Error("asset1ID and asset2ID must be different");
  }

  // If I pass NaN to an asset id, Math.min or Math.max will always return the NaN, this check prevents it too.
  if (asset1ID < 0 || asset2ID < 0) {
    throw new Error("asset1ID and asset2ID must be positive integer");
  }
```

image

File: src/validator.test.ts (hunk 1 above)
Comment (Anlerkan) [reply src/validator.test.ts:15]: We simply access a field of an object and return it. We will use the same object even if we have that test so I think we do not need that test.

**Review by Anlerkan — COMMENTED** [review]
Thanks @jamcry, looks good.

File: src/validator.test.ts (hunk 1 above)
Comment (edizcelik) [reply src/validator.test.ts:15]: I think it doesn't hurt to have it and offers some safety

**Review by edizcelik — APPROVED** [review]

File: src/util/asset/assetUtils.test.ts (hunk 3 above)
Comment (jamcry) [reply src/util/asset/assetUtils.test.ts:6]: yeah It can be nice, I'll create a ticket for this

**Pull request merged**